src/domain/ai.py

```
@@ -0,0 +1,406 @@
+        for chunk in stream:
+            if chunk.choices[0].delta.content is not None:
+                response_text += chunk.choices[0].delta.content
+
+                # Call on_chunk callback periodically
+                if on_chunk:
+                    current_time = time.time()
+                    if current_time - last_update_time > update_interval:
+                        try:
+                            await _call_callback(on_chunk, response_text)
+                            last_update_time = current_time
+                        except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)